flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1832,6 +1837,16 @@ IntrinsicLibrary::genCount(mlir::Type resultType,
       });
 }
 
+// CPU_TIME
+void IntrinsicLibrary::genCpuTime(llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 1);
+  auto arg = args[0].getUnboxed();
```

Comment:
Even though this should be guaranteed by the front-end and lowering, I would still add a small `assert(arg && "time must be a scalar ")` to rule-out any undefined behaviors.